Add a table with drop-down editors features (#1728)
DevExpress/devextreme-documentation: concepts/05 UI Components/SelectBox/How to Choose a Drop-Down Editor.md

DevExtreme has five drop-down text editors:

- [SelectBox](/Documentation/ApiReference/UI_Components/dxSelectBox/)
- [TagBox](/Documentation/ApiReference/UI_Components/dxTagBox/)
- [Lookup](/Documentation/ApiReference/UI_Components/dxLookup/)
- [Autocomplete](/Documentation/ApiReference/UI_Components/dxAutocomplete/)
- [DropDownBox](/Documentation/ApiReference/UI_Components/dxDropDownBox/)

> The following table lists available drop-down editor features:

Reviewer: ```suggestion
The following table compares the drop-down editors by a number of features:
```

        <td>-</td>
        <td>Yes</td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxDropDownBox/Configuration/#acceptCustomValue">Yes</a></td>
    </tr>
    <tr>
        <td>Custom buttons</td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/buttons/">Yes</a></td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/buttons/">Yes</a></td>
        <td>-</td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxAutocomplete/Configuration/buttons/">Yes</a></td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxDropDownBox/Configuration/buttons/">Yes</a></td>
    </tr>
    <tr>
        <td>Keyboard navigation</td>
        <td><a href="/Documentation/Guide/UI_Components/SelectBox/Keyboard_Support/">Yes</a></td>
        <td><a href="/Documentation/Guide/UI_Components/TagBox/Keyboard_Support/">Yes</a></td>
        <td><a href="/Documentation/Guide/UI_Components/Lookup/Keyboard_Support/">Yes</a></td>
        <td>Yes</td>
        <td>Requires custom configuration</td>
    </tr>
    <tr>

>         <td>Search box in drop down</td>

Reviewer: ```suggestion
        <td>Search box in a drop-down field</td>
```

DevExtreme has five drop-down text editors:

- [SelectBox](/Documentation/ApiReference/UI_Components/dxSelectBox/)
- [TagBox](/Documentation/ApiReference/UI_Components/dxTagBox/)
- [Lookup](/Documentation/ApiReference/UI_Components/dxLookup/)
- [Autocomplete](/Documentation/ApiReference/UI_Components/dxAutocomplete/)
- [DropDownBox](/Documentation/ApiReference/UI_Components/dxDropDownBox/)

> The following table compares the drop-down editors by a number of features:

Reviewer 2: ```suggestion
The following table compares the drop-down editors according to features:
```

        <td><a href="/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#minSearchLength">Yes</a></td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#minSearchLength">Yes</a></td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxAutocomplete/Configuration/#minSearchLength">Yes</a></td>
        <td>Requires custom configuration</td>
    </tr>
    <tr>
        <td>Multiple selection</td>
        <td>-</td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#showSelectionControls">Yes</a></td>
        <td>-</td>
        <td>-</td>
        <td>Requires custom configuration</td>
    </tr>
    <tr>
        <td>Paging</td>
        <td><a href="/Documentation/Guide/UI_Components/SelectBox/Enable_Paging/">Yes</a></td>
        <td><a href="/Documentation/Guide/UI_Components/TagBox/Enable_Paging/">Yes</a></td>
        <td><a href="/Documentation/Guide/UI_Components/Lookup/Enable_Paging/">Yes</a></td>
        <td>-</td>
        <td>Requires custom configuration</a></td>
    </tr>
</table>

> Unlike the rest of the editors, DropDownBox allows you to put any content in its drop-down field. This feature makes it more versatile but at the same time requires more effort from the developer. We recommend using DropDownBox in those advanced use cases that cannot be satisfied by other editors. Refer to the DropDownBox [help topics](/Documentation/Guide/UI_Components/DropDownBox/Overview/) and [demos](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DropDownBox/SingleSelection/) for further information.

Reviewer 2: DropDownBox allows you to put any content in its drop-down field. This feature makes it more versatile but it is also more complex to implement. 

We recommend that you use DropDownBox in cases **that cannot be satisfied by other editors** - Why do you use this phrase? How are they not **satisfied** by other editors? Which cases do you refer to?

Author: Hello. The DDBox is basically a container for other components, and this allows users implement almost any scenario they can imagine. For instance, they want display two columns with text and/or images in the drop-down field. They can nest our dxDataGrid component into the drop-down field, and our DDBox is the only component among other drop-down editors that has such capabilities.

Reviewer 2: I think a detailed explanation, that is, something like 'DropDownBox allows you to put any content in its drop-down field. This feature makes it more versatile but it is also more complex to implement' (+ perhaps additional information) already explains why you would use this instead of other editors. You can even incorporate an example of such scenario in the description to help illustrate what makes it more versatile. Saying something is better does not mean anything unless you explain why it is better.

#####See Also#####

> - [Lookup vs SelectBox](/Documentation/Guide/UI_Components/Lookup/Lookup_vs_SelectBox/)

Reviewer: Shouldn't we also refer to this article from Lookup vs SelectBox?

        <td><a href="/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#minSearchLength">Yes</a></td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#minSearchLength">Yes</a></td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxAutocomplete/Configuration/#minSearchLength">Yes</a></td>
        <td>Requires custom configuration</td>
    </tr>
    <tr>
        <td>Multiple selection</td>
        <td>-</td>
        <td><a href="/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#showSelectionControls">Yes</a></td>
        <td>-</td>
        <td>-</td>
        <td>Requires custom configuration</td>
    </tr>
    <tr>
        <td>Paging</td>
        <td><a href="/Documentation/Guide/UI_Components/SelectBox/Enable_Paging/">Yes</a></td>
        <td><a href="/Documentation/Guide/UI_Components/TagBox/Enable_Paging/">Yes</a></td>
        <td><a href="/Documentation/Guide/UI_Components/Lookup/Enable_Paging/">Yes</a></td>
        <td>-</td>
        <td>Requires custom configuration</a></td>
    </tr>
</table>

> The DropDownBox UI component's API allows you to display custom content in its drop-down field. This API requires that you manually implement other components within DropDownBox and their features (see [dropDownOptions](/Documentation/ApiReference/UI_Components/dxDropDownBox/Configuration/#dropDownOptions). 
> 
> Refer to the [guides](/Documentation/Guide/UI_Components/DropDownBox/Overview/) and [demos](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DropDownBox/SingleSelection/) sections for the DropDownBox to see more available customizations. 

Reviewer: > The DropDownBox UI component's API allows you to display custom content in its drop-down field. This API requires that you manually implement other components within DropDownBox and their features (see [dropDownOptions](/Documentation/ApiReference/UI_Components/dxDropDownBox/Configuration/#dropDownOptions). 

These are just facts. They are not good, nor bad. But in this article, simple facts won't do. We need to compare editors with each other, give some recommendations, and substantiate them.

```suggestion
Unlike the rest of the editors, DropDownBox allows you to put any content in its drop-down field. This feature makes it more versatile but at the same time requires more effort from the developer. We recommend using DropDownBox in those advanced use cases that cannot be satisfied by other editors. Refer to the DropDownBox [help topics](/Documentation/Guide/UI_Components/DropDownBox/Overview/) and [demos](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DropDownBox/SingleSelection/) for further information.
```
